lager/extra/qt.hpp

```diff
@@ -0,0 +1,85 @@
+//
+// lager - library for functional interactive c++ programs
+// Copyright (C) 2017 Juan Pedro Bolivar Puente
+//
+// This file is part of lager.
+//
+// lager is free software: you can redistribute it and/or modify
+// it under the terms of the MIT License, as detailed in the LICENSE
+// file located at the root of this source code distribution,
+// or here: <https://github.com/arximboldi/lager/blob/master/LICENSE>
+//
+
+#pragma once
+
+#include <lager/cursor.hpp>
+#include <lager/reader.hpp>
+#include <lager/watch.hpp>
+
+namespace lager {
+
+namespace detail {
+
+struct qt_helper
+{};
+
+} // namespace detail
+
+/*!
+ * Use in a class declaration before any FunQ definition.
+ */
+#define LAGER_QT_ENABLE(klass)                                                 \
+    using LAGER_QT_CURRENT_CLASS = klass;                                      \
+    /**/
+
+/*!
+ * Return the name of the underlying value supporting a FunQ
+ * property.
+ */
+#define LAGER_QT(name) lager_qt__##name##__
+
+/*!
+ * Use in a QObject class declaration to define a property supported
+ * by a single member that is not expected to change.
+ */
+#define LAGER_QT_CONST(type, name)                                             \
+    type LAGER_QT(name);                                                       \
+    Q_PROPERTY(type name READ name CONSTANT)                                   \
+    const type& name() const { return LAGER_QT(name); }                        \
+    /**/
+
+#define LAGER_QT_SIGNAL_DETAIL(type, name)                                     \
+    Q_SIGNAL void name##Changed(const type& name);                             \
+    ::lager::detail::qt_helper funq__##name##__initHelper__ = [this] {         \
+        ::lager::watch(LAGER_QT(name),                                         \
+                       [this](const type& old, const type& curr) {             \
```

Comment:
`old` will produce a warning an unused variable here

Reply:
Not sure I care :p

Reply:
😱